vscode/src/completions/inline-completion-item-provider.ts

```diff
@@ -489,6 +500,50 @@ export class InlineCompletionItemProvider implements vscode.InlineCompletionItem
         )
     }
 
+    public async formatCompletion(autocompleteItem: AutocompleteItem): Promise<void> {
+        try {
+            const {
+                document,
+                position,
+                docContext: { currentLinePrefix },
+            } = autocompleteItem.requestParams
+
+            const insertedLines = lines(autocompleteItem.analyticsItem.insertText)
+            const endPosition =
+                insertedLines.length <= 1
+                    ? new vscode.Position(position.line, currentLinePrefix.length + insertedLines[0].length)
+                    : new vscode.Position(position.line + insertedLines.length - 1, insertedLines.at(-1)!.length)
+            const rangeToFormat = new vscode.Range(position, endPosition)
+
+            const formattingChanges = await vscode.commands.executeCommand<vscode.TextEdit[] | undefined>(
+                'vscode.executeFormatRangeProvider',
+                document.uri,
+                rangeToFormat,
+                {
+                    tabSize: getEditorTabSize(document.uri),
+                    insertSpaces: getEditorInsertSpaces(document.uri),
+                }
+            )
+
+            const formattingChangesInRange = (formattingChanges || []).filter(change =>
+                rangeToFormat.contains(change.range)
+            )
+
+            if (formattingChangesInRange.length !== 0) {
+                const edit = new vscode.WorkspaceEdit()
+                for (const change of formattingChangesInRange) {
+                    edit.replace(document.uri, change.range, change.newText)
+                }
+                void vscode.workspace.applyEdit(edit)
+            }
+        } catch (unknownError) {
+            const error = unknownError instanceof Error ? unknownError : new Error('unknown')
```

Comment:
```suggestion
            const error = unknownError instanceof Error ? unknownError : new Error(unknownError ?? 'unknown')
```

Would that work in case it has a toString?